cpp/include/cudf/timezone.hpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/table/table.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+
+#include <memory>
+#include <optional>
+#include <string>
+
+namespace cudf {
+
+// Cycle in which the time offsets repeat
+static constexpr int32_t cycle_years = 400;
+// Number of future entries in the timezone transition table:
+// Two entries per year, over the length of the cycle
+static constexpr uint32_t cycle_entry_cnt = 2 * cycle_years;
+
+/**
+ * @brief Creates a transition table to convert ORC timestamps to UTC.
+ *
+ * Uses system's TZif files. Assumes little-endian platform when parsing these files.
+ *
+ * @param tzif_dir The directory where the TZif files are located
+ * @param timezone_name standard timezone name (for example, "US/Pacific")
```

Comment:
Let's not use an obsolete time zone for our example. `US/Pacific` has been replaced by `America/Los_Angeles`.

http://www.timezoneconverter.com/cgi-bin/zoneinfo.tzc?s=default&tz=US/Pacific